File: CNC modeling 2.0/src/com/sergey/pisarev/interfaces/PresenterImpl.java
package com.sergey.pisarev.interfaces;

import javafx.scene.input.DragEvent;

public interface PresenterImpl {
    void onStart();
    void onCycleStart();
    void onSingleBlock();
    void onReset();
    void onMouseClickedProgram(int numberLine);
    void openDragProgram(DragEvent event);
    void openDragParameter(DragEvent event);
}

File: CNC modeling 2.0/src/main/java/com/sergey/pisarev/interfaces/PresenterImpl.kt
package com.sergey.pisarev.interfaces

import javafx.scene.input.DragEvent
import javafx.scene.input.MouseEvent
import javafx.scene.input.ScrollEvent

interface PresenterImpl {
    fun onStart(program: String?)
    fun onCycleStart(program: String?, isSelectToolRadius: Boolean?)
    fun onSingleBlock(isClick: Boolean)
    fun onReset()
    fun getCaretPosition(numberLine: Int)
    fun openDragProgram(event: DragEvent?)
    fun setOnChangesTextProgram(program: String?)
    fun onMouseClickedCanvas(event: MouseEvent?)
    fun onMouseMovedCanvas(event: MouseEvent?)
    fun handleZooming(event: ScrollEvent?)
    fun handleMousePressed(event: MouseEvent?)
    fun handleMouseDragged(event: MouseEvent?)
    fun initSystemCoordinate(canvasWidth: Double, canvasHeight: Double)
    fun convertAviaProgram(aviaProgram: String?)
    fun convertTheProgramToG17(program: String?)
    fun convertTheProgramToG18(program: String?)
    fun checkChangesProgram(program: String?)
    fun saveProgram(program: String?)
}